src/wheel/_wheelfile.py

```diff
@@ -0,0 +1,515 @@
+                        "file?"
+                    )
+        except BaseException:
+            self._zip.close()
+            raise
+
+        self._dist_info_dir = f"{self.name}-{self.version}.dist-info"
+        self._data_dir = f"{self.name}-{self.version}.data"
+        self._record_entries = self._read_record()
+        return self
+
+    def __exit__(
+        self,
+        exc_type: type[BaseException],
+        exc_val: BaseException,
+        exc_tb: TracebackType,
+    ) -> None:
+        self._zip.close()
+        self._record_entries.clear()
+
+    def _read_record(self) -> OrderedDict[str, WheelRecordEntry]:
+        entries = OrderedDict()
+        contents = self.read_dist_info("RECORD")
+        reader = csv.reader(
+            contents.strip().split("\n"),
+            delimiter=",",
+            quotechar='"',
+            lineterminator="\n",
+        )
+        for row in reader:
+            if not row:
+                break
+
+            path, hash_digest, filesize = row
+            if hash_digest:
+                algorithm, hash_digest = hash_digest.split("=")
+                try:
+                    hashlib.new(algorithm)
+                except ValueError:
+                    raise WheelError(
+                        f"Unsupported hash algorithm: {algorithm}"
+                    ) from None
+
+                if algorithm.lower() in {"md5", "sha1"}:
+                    raise WheelError(
+                        f"Weak hash algorithm ({algorithm}) is not permitted by PEP 427"
+                    )
+
+                entries[path] = WheelRecordEntry(
+                    algorithm, _decode_hash_value(hash_digest), int(filesize)
+                )
+
+        return entries
+
+    @property
+    def dist_info_dir(self) -> str:
+        return self._dist_info_dir
+
+    @property
+    def data_dir(self) -> str:
+        return self._data_dir
+
+    @property
+    def dist_info_filenames(self) -> list[PurePath]:
+        return [
+            PurePath(fname)
+            for fname in self._zip.namelist()
+            if fname.startswith(self._dist_info_dir)
+        ]
+
+    @property
+    def filenames(self) -> list[PurePath]:
+        return [PurePath(fname) for fname in self._zip.namelist()]
+
+    def read_dist_info(self, filename: str) -> str:
+        filename = self.dist_info_dir + "/" + filename
+        try:
+            contents = self._zip.read(filename)
+        except KeyError:
+            raise WheelError(f"File {filename!r} not found") from None
+
+        return contents.decode("utf-8")
+
+    def get_contents(self) -> Iterator[WheelContentElement]:
+        for fname, entry in self._record_entries.items():
+            with self._zip.open(fname, "r") as stream:
+                yield WheelContentElement(
+                    PurePath(fname), entry.hash_value, entry.filesize, stream
+                )
+
+    def test(self) -> None:
+        """Verify the integrity of the contained files."""
```

Comment:
```suggestion
    def validate_record(self) -> None:
        """Verify the integrity of the contained files."""
```

This is a more complete name IMO. :)

It also matches the suggestion in https://github.com/pypa/installer/pull/147/. :)